lxc/init.go

```diff
@@ -409,6 +409,7 @@ func (c *cmdInit) create(conf *config.Config, args []string) (lxd.InstanceServer
 	if len(instances) == 1 && name == "" {
 		fields := strings.Split(instances[0], "/")
 		name = fields[len(fields)-1]
+		name = strings.SplitN(name, "?", 2)[0]
```

Comment:
Where does the `?` get added?

Reply:
From `operation > resources`. Operation is returned from `CreateInstanceFromImage()`.

Resource looks like this:
```
/1.0/instances/knowing-rabbit?project=test
```

Currently the resource is separated by `/` and the last part is taken, which in the above case results in `knowing-rabbit?project=test`, so I just removed the query part.

Reply:
Ah OK so these are the sort of bugs that occur when we try parsing the URL manually rather than using something `url.Parse()` which I think we should use here instead, as it will also avoid issues like URL segment and query string escaping issues.

Reply:
There's been quite a lot of this in the past in the LXD code base and something I'd like to see us move away from.

Reply:
Agree :)